src/test/java/org/opensearch/security/hasher/BCryptPasswordHasherTests.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.hasher;
+
+import java.nio.CharBuffer;
+
+import org.junit.Test;
+
+import org.opensearch.OpenSearchSecurityException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThrows;
+
+public class BCryptPasswordHasherTests {
+
+    private final PasswordHasher passwordHasher = new BCryptPasswordHasher();
+
+    private final String password = "testPassword";
+    private final String wrongPassword = "wrongTestPassword";
+
+    @Test
+    public void shouldMatchHashToCorrectPassword() {
+        String hashedPassword = passwordHasher.hash(password.toCharArray());
+        assertThat(passwordHasher.check(password.toCharArray(), hashedPassword), is(true));
+    }
+
+    @Test
+    public void shouldNotMatchHashToWrongPassword() {
+        String hashedPassword = passwordHasher.hash(password.toCharArray());
+        assertThat(passwordHasher.check(wrongPassword.toCharArray(), hashedPassword), is(false));
+
+    }
+
+    /**
+     * Ensures that the hashes that were previously created by OpenBSDBCrypt are still valid
+     */
+    @Test
+    public void shouldBeBackwardsCompatible() {
+        String legacyHash = "$2y$12$gdh2ecVBQmwpmcAeyReicuNtXyR6GMWSfXHxtcBBqFeFz2VQ8kDZe";
```

Comment:
For my learning where did you get this from? Is this hard coded somewhere as the default legacy hash used by Bcrypt?

Reply:
nit: we should document how this has is constructed - to keep track of this test for further references.

Reply:
@derek-ho I just used OpenBSDBCrypt to hash "password" with the default parameters. 

In BCrypt every hash will be unique for the same input. However, every implementation of BCrypt should be able to verify any BCrypt hashes when specifying the same parameters. (In this case, if you look at the prefix $2y$12 it tells you that it was generated with the "2y" version of the algorithm with 12 log rounds)

Reply:
> nit: we should document how this has is constructed - to keep track of this test for further references.

I added a comment above the unit test saying "Ensures that the hashes that were previously created by OpenBSDBCrypt are still valid". Should I clarify more?

Reply:
Thanks, looks good for now!